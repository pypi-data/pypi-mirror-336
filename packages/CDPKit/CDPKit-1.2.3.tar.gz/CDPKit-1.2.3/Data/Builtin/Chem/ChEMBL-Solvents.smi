WATER	[OH2]
DICHLOROMETHANE	ClCCl
TRICHLOROMETHANE	ClC(Cl)Cl
ETHYL ACETATE	CCOC(=O)C
METHANOL	CO
PROPAN-2-OL	CC(C)O
ACETONE	CC(=O)C
DMSO	CS(=O)C
ETHANOL	CCO
