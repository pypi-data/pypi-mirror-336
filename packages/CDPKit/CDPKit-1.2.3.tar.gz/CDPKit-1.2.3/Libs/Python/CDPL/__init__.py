##
# __init__.py  
#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) 2003 Thomas Seidel <thomas.seidel@univie.ac.at>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program; see the file COPYING. If not, write to
# the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
# Boston, MA 02111-1307, USA.
##


from __future__ import absolute_import

import os
import sys
import platform

if sys.version_info[1] >= 8 and platform.system() == 'Windows':
    if 'site-packages' in __file__:
        os.add_dll_directory(os.path.dirname(__file__))
    else:
        os.add_dll_directory(os.path.join(os.path.dirname(os.path.dirname(os.path.dirname(__file__))), '@CDPKIT_EXECUTABLE_INSTALL_DIR@'))

from ._cdpl import *

__version__ = VERSION_STRING
