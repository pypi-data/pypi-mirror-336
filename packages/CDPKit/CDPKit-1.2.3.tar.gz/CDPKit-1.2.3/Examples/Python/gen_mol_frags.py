#!/bin/env python

##
# gen_mol_frags.py 
#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) 2003 Thomas Seidel <thomas.seidel@univie.ac.at>
#
# Permission to use, copy, modify, and/or distribute this software for any purpose with
# or without fee is hereby granted.
# THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH REGARD
# TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS. IN
# NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL
# DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER
# IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
# CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
##


import sys
import argparse

import CDPL.Chem as Chem

    
# splits a given input molecule into fragments using the provided Chem.FragmentGenerator instance implementing
# the fragmentation rules
def genFragments(frag_gen: Chem.FragmentGenerator, in_mol: Chem.Molecule, out_frags: Chem.FragmentList) -> None:
    Chem.initSubstructureSearchTarget(in_mol, False) # calculate required properties

    frag_gen.generate(in_mol, out_frags)

def getLogMessage(verb_level: int, mol_id: str, num_frags: int, frag_gen: Chem.FragmentGenerator) -> str:
    if verb_level == 0:
        return None

    if num_frags <= 1:
        return  f'- Molecule {mol_id}: no fragmentation'

    msg = f'- Molecule {mol_id}: {num_frags} fragments'

    if verb_level < 2:
        return msg

    msg += '\n  Cut bonds (Rule ID, Bond index, Atom#1 index, Atom#2 index, Fragment#1 index, Fragment#2 index):'
    
    # gather information about the bonds that were split during fragmentation
    for i in range(frag_gen.numFragmentLinks):
        frag_link = frag_gen.getFragmentLink(i)
        msg += (f'\n  {frag_link.ruleID}, {frag_link.bond.index}, {frag_link.bond.begin.index}, '
                f'{frag_link.bond.end.index}, {frag_link.fragment1Index}, {frag_link.fragment2Index}')
    
    return msg

def parseArgs() -> argparse.Namespace:
    parser = argparse.ArgumentParser(description='Performs molecule fragmentation using BRICS or RECAP rules.')

    parser.add_argument('-i',
                        dest='in_file',
                        required=True,
                        metavar='<file>',
                        help='Input molecule file')
    parser.add_argument('-o',
                        dest='out_file',
                        required=True,
                        metavar='<file>',
                        help='Fragment output file')
    parser.add_argument('-r',
                        dest='rule_set',
                        required=True,
                        metavar='<string>',
                        help='Fragmentation rule set (BRICS or RECAP)')
    parser.add_argument('-v',
                        dest='verb_level',
                        required=False,
                        metavar='<0|1|2>',
                        choices=range(0, 3),
                        default=1,
                        help='Verbosity level (default: 1; 0 -> no console output, 1 -> verbose, 2 -> extra verbose)',
                        type=int)
    parser.add_argument('-m',
                        dest='output_mol',
                        required=False,
                        action='store_true',
                        default=False,
                        help='Output the fragmented molecule before its fragments (default: false)')
    parser.add_argument('-b',
                        dest='inc_split_bonds',
                        required=False,
                        action='store_true',
                        default=False,
                        help='Include bonds that were split during fragmentation in the obtained fragments (default: false)')
    parser.add_argument('-x',
                        dest='atom_mpg',
                        required=False,
                        action='store_true',
                        default=False,
                        help='Label atoms with atom mapping IDs (default: false)')

    return parser.parse_args()

def main() -> None:
    args = parseArgs() # process command line arguments
    
    # create an instance of the class implementing the desired fragmentation rule set
    if args.rule_set.lower() == "brics":
        frag_gen = Chem.BRICSFragmentGenerator()
    elif args.rule_set.lower() == "recap":
        frag_gen = Chem.RECAPFragmentGenerator()
    else:
        sys.exit('Error: invalid fragmentation rule set')

    frag_gen.includeSplitBonds(args.inc_split_bonds) # apply -b option
        
    # create reader for input molecules (format specified by file extension)
    reader = Chem.MoleculeReader(args.in_file) 

    # create writer for output fragments (format specified by file extension)
    writer = Chem.MolecularGraphWriter(args.out_file) 
  
    if args.atom_mpg: # enable atom mapping IDs for SMILES output
        Chem.setSMILESMolOutputAtomMappingIDParameter(writer, True)
    
    # create instances of the default implementation of the Chem.Molecule interface for the input molecule
    in_mol = Chem.BasicMolecule()

    # create instances of the Chem.FragmentList data structure storing the generated fragments
    out_frags = Chem.FragmentList()
        
    i = 1
    
    try:
        # read and process molecules one after the other until the end of input has been reached (or a severe error occurs)
        while reader.read(in_mol):
            # compose a molecule identifier
            mol_name = Chem.getName(in_mol).strip() 

            if mol_name == '':
                mol_name = f'Mol#{i}'
                mol_id = f'#{i}'  # fallback if name is empty or not available
            else:
                mol_id = f'\'{mol_name}\' (#{i})'
         
            try:
                # perform molecule fragmentation
                genFragments(frag_gen, in_mol, out_frags) 

                log_msg = getLogMessage(args.verb_level, mol_id, len(out_frags), frag_gen)

                if log_msg:
                    print(log_msg)

                if args.atom_mpg: # label atoms with atom mapping IDs to see correspondences between fragment and parent molecule atoms
                    for atom in in_mol.atoms:
                        Chem.setAtomMappingID(atom, atom.getIndex() + 1)

                try:
                    if args.output_mol: # write input molecule before the generated fragments
                        if not writer.write(in_mol):
                            sys.exit(f'Error: writing molecule {mol_id} failed')

                    for j in range(len(out_frags)):
                        frag = out_frags[j]

                        # set fragment name encoding the parent molecule and fragment number
                        Chem.setName(frag, mol_name + f'_F#{j + 1}') 
                        
                        # calculate (if not present) some basic properties of the output fragments
                        # that might be required for writing (output format dependent)
                        Chem.perceiveSSSR(frag, False)
                        Chem.perceiveComponents(frag, False)

                        try:
                            # write fragment
                            if not writer.write(frag):
                                sys.exit('Error: writing fragment of molecule {mol_id} failed')
                        except Exception as e: # handle exception raised in case of severe write errors
                             sys.exit(f'Error: writing fragment of molecule {mol_id} failed: {str(e)}')
                             
                except Exception as e: # handle exception raised in case of severe write errors
                    sys.exit(f'Error: writing molecule {mol_id} failed: {str(e)}')
                
            except Exception as e: # handle exception raised in case of severe structure processing errors
                sys.exit(f'Error: processing of molecule {mol_id} failed: {str(e)}')

            i += 1
            
    except Exception as e: # handle exception raised in case of severe read errors
        sys.exit(f'Error: reading of molecule {mol_id} failed: {str(e)}')

    writer.close()
    sys.exit(0)
        
if __name__ == '__main__':
    main()
