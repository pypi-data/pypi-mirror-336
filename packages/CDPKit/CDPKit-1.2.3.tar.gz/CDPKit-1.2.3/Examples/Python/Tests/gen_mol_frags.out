[c:1]1([c:2]([c:3]([c:4]2[c:5]([c:6]1[H:7])\[C:8]([C:9]([N:10]2[H:11])=[O:12])=[C:13](\[N:14]([c:15]2[c:16]([c:17]([c:18]([c:19]([c:20]2[H:21])[H:22])[S@@:23]([N:24]([C:25]([H:26])([H:27])[H:28])[H:29])(=[O:30])=[O:31])[H:32])[H:33])[H:34])[H:35])[H:36])[H:37])[H:38] ZINC03814443|cdk2_ligands|dock1
[c:1]1([c:2]([c:3]([c:4]2[c:5]([c:6]1[H:7])\[C:8]([C:9]([N:10]2[H:11])=[O:12])=[C:13](\[N:14]([cH3:15])[H:34])[H:35])[H:36])[H:37])[H:38] ZINC03814443|cdk2_ligands|dock1_F#1
[c:15]1([NH2:14])[c:16]([c:17]([c:18]([c:19]([c:20]1[H:21])[H:22])[S@@:23]([NH2:24])(=[O:30])=[O:31])[H:32])[H:33] ZINC03814443|cdk2_ligands|dock1_F#2
[N:24]([SH:23])([C:25]([H:26])([H:27])[H:28])[H:29] ZINC03814443|cdk2_ligands|dock1_F#3
[c:1]1([c:2]([c:3]([c:4]2[c:5]([c:6]1[H:7])\[C:8]([C:9]([N:10]2[H:11])=[O:12])=[C:13](\[N:14]([c:15]2[c:16]([c:17]([c:18]([c:19]([c:20]2[H:21])[H:22])[S@@:23]([N:24]([C:25]([H:26])([H:27])[H:28])[H:29])(=[O:30])=[O:31])[H:32])[H:33])[H:34])[H:35])[H:36])[H:37])[H:38] ZINC03814443|cdk2_ligands|dock5
[c:1]1([c:2]([c:3]([c:4]2[c:5]([c:6]1[H:7])\[C:8]([C:9]([N:10]2[H:11])=[O:12])=[C:13](\[N:14]([cH3:15])[H:34])[H:35])[H:36])[H:37])[H:38] ZINC03814443|cdk2_ligands|dock5_F#1
[c:15]1([NH2:14])[c:16]([c:17]([c:18]([c:19]([c:20]1[H:21])[H:22])[S@@:23]([NH2:24])(=[O:30])=[O:31])[H:32])[H:33] ZINC03814443|cdk2_ligands|dock5_F#2
[N:24]([SH:23])([C:25]([H:26])([H:27])[H:28])[H:29] ZINC03814443|cdk2_ligands|dock5_F#3
[c:1]1([c:2]([c:3]([c:4]2[c:5]([c:6]1[H:7])\[C:8]([C:9]([N:10]2[H:11])=[O:12])=[C:13](\[N:14]([c:15]2[c:16]([c:17]([c:18]([c:19]([c:20]2[H:21])[H:22])[S@@:23]([N:24]([C:25]([H:26])([H:27])[H:28])[H:29])(=[O:30])=[O:31])[H:32])[H:33])[H:34])[H:35])[H:36])[H:37])[H:38] ZINC03814443|cdk2_ligands|dock9
[c:1]1([c:2]([c:3]([c:4]2[c:5]([c:6]1[H:7])\[C:8]([C:9]([N:10]2[H:11])=[O:12])=[C:13](\[N:14]([cH3:15])[H:34])[H:35])[H:36])[H:37])[H:38] ZINC03814443|cdk2_ligands|dock9_F#1
[c:15]1([NH2:14])[c:16]([c:17]([c:18]([c:19]([c:20]1[H:21])[H:22])[S@@:23]([NH2:24])(=[O:30])=[O:31])[H:32])[H:33] ZINC03814443|cdk2_ligands|dock9_F#2
[N:24]([SH:23])([C:25]([H:26])([H:27])[H:28])[H:29] ZINC03814443|cdk2_ligands|dock9_F#3
[c:1]1([c:2]([c:3]([c:4]2[c:5]([c:6]1[H:7])\[C:8]([C:9]([N:10]2[H:11])=[O:12])=[C:13](\[N:14]([c:15]2[c:16]([c:17]([c:18]([c:19]([c:20]2[H:21])[H:22])[S@:23]([N:24]([c:25]2[n:26][c:27]([c:28]([s:29]2)[H:30])[H:31])[H:32])(=[O:33])=[O:34])[H:35])[H:36])[H:37])[H:38])[H:39])[H:40])[H:41] ZINC03814447|cdk2_ligands|dock1
[c:1]1([c:2]([c:3]([c:4]2[c:5]([c:6]1[H:7])\[C:8]([C:9]([N:10]2[H:11])=[O:12])=[C:13](\[N:14]([cH3:15])[H:37])[H:38])[H:39])[H:40])[H:41] ZINC03814447|cdk2_ligands|dock1_F#1
[c:15]1([NH2:14])[c:16]([c:17]([c:18]([c:19]([c:20]1[H:21])[H:22])[S@:23]([NH2:24])(=[O:33])=[O:34])[H:35])[H:36] ZINC03814447|cdk2_ligands|dock1_F#2
[N:24]([SH:23])([cH3:25])[H:32] ZINC03814447|cdk2_ligands|dock1_F#3
[c:25]1([NH2:24])[n:26][c:27]([c:28]([s:29]1)[H:30])[H:31] ZINC03814447|cdk2_ligands|dock1_F#4
[c:1]1([c:2]([c:3]([c:4]2[c:5]([c:6]1[H:7])\[C:8]([C:9]([N:10]2[H:11])=[O:12])=[C:13](\[N:14]([c:15]2[c:16]([c:17]([c:18]([c:19]([c:20]2[H:21])[H:22])[S@:23]([N:24]([c:25]2[n:26][c:27]([c:28]([s:29]2)[H:30])[H:31])[H:32])(=[O:33])=[O:34])[H:35])[H:36])[H:37])[H:38])[H:39])[H:40])[H:41] ZINC03814447|cdk2_ligands|dock4
[c:1]1([c:2]([c:3]([c:4]2[c:5]([c:6]1[H:7])\[C:8]([C:9]([N:10]2[H:11])=[O:12])=[C:13](\[N:14]([cH3:15])[H:37])[H:38])[H:39])[H:40])[H:41] ZINC03814447|cdk2_ligands|dock4_F#1
[c:15]1([NH2:14])[c:16]([c:17]([c:18]([c:19]([c:20]1[H:21])[H:22])[S@:23]([NH2:24])(=[O:33])=[O:34])[H:35])[H:36] ZINC03814447|cdk2_ligands|dock4_F#2
[N:24]([SH:23])([cH3:25])[H:32] ZINC03814447|cdk2_ligands|dock4_F#3
[c:25]1([NH2:24])[n:26][c:27]([c:28]([s:29]1)[H:30])[H:31] ZINC03814447|cdk2_ligands|dock4_F#4
[c:1]1([c:2]([c:3]([c:4]2[c:5]([c:6]1[H:7])\[C:8]([C:9]([N:10]2[H:11])=[O:12])=[C:13](\[N:14]([c:15]2[c:16]([c:17]([c:18]([c:19]([c:20]2[H:21])[H:22])[S@:23]([N:24]([c:25]2[n:26][c:27]([c:28]([s:29]2)[H:30])[H:31])[H:32])(=[O:33])=[O:34])[H:35])[H:36])[H:37])[H:38])[H:39])[H:40])[H:41] ZINC03814447|cdk2_ligands|dock2
[c:1]1([c:2]([c:3]([c:4]2[c:5]([c:6]1[H:7])\[C:8]([C:9]([N:10]2[H:11])=[O:12])=[C:13](\[N:14]([cH3:15])[H:37])[H:38])[H:39])[H:40])[H:41] ZINC03814447|cdk2_ligands|dock2_F#1
[c:15]1([NH2:14])[c:16]([c:17]([c:18]([c:19]([c:20]1[H:21])[H:22])[S@:23]([NH2:24])(=[O:33])=[O:34])[H:35])[H:36] ZINC03814447|cdk2_ligands|dock2_F#2
[N:24]([SH:23])([cH3:25])[H:32] ZINC03814447|cdk2_ligands|dock2_F#3
[c:25]1([NH2:24])[n:26][c:27]([c:28]([s:29]1)[H:30])[H:31] ZINC03814447|cdk2_ligands|dock2_F#4
[c:1]1([c:2]([c:3]([c:4]2[c:5]([c:6]1[H:7])[C:8](\[C:9](=[C:10]1/[c:11]3[c:12]([c:13]([c:14]([c:15]([c:16]3[N:17]([C:18]1=[O:19])[H:20])[H:21])[H:22])[S:23](=[O:24])(=[O:25])[O-:26])[H:27])[N:28]2[H:29])=[O:30])[H:31])[H:32])[H:33] ZINC03814440|cdk2_ligands|dock3
[c:1]1([c:2]([c:3]([c:4]2[c:5]([c:6]1[H:7])[C:8](\[C:9](=[C:10]1/[c:11]3[c:12]([c:13]([c:14]([c:15]([c:16]3[N:17]([C:18]1=[O:19])[H:20])[H:21])[H:22])[S:23](=[O:24])(=[O:25])[O-:26])[H:27])[N:28]2[H:29])=[O:30])[H:31])[H:32])[H:33] ZINC03814440|cdk2_ligands|dock3_F#1
[c:1]1([c:2]([c:3]([c:4]2[c:5]([c:6]1[H:7])[C:8](\[C:9](=[C:10]1/[c:11]3[c:12]([c:13]([c:14]([c:15]([c:16]3[N:17]([C:18]1=[O:19])[H:20])[H:21])[H:22])[S:23](=[O:24])(=[O:25])[O-:26])[H:27])[N:28]2[H:29])=[O:30])[H:31])[H:32])[H:33] ZINC03814440|cdk2_ligands|dock1
[c:1]1([c:2]([c:3]([c:4]2[c:5]([c:6]1[H:7])[C:8](\[C:9](=[C:10]1/[c:11]3[c:12]([c:13]([c:14]([c:15]([c:16]3[N:17]([C:18]1=[O:19])[H:20])[H:21])[H:22])[S:23](=[O:24])(=[O:25])[O-:26])[H:27])[N:28]2[H:29])=[O:30])[H:31])[H:32])[H:33] ZINC03814440|cdk2_ligands|dock1_F#1
[c:1]1([c:2]([c:3]([c:4]2[c:5]([c:6]1[H:7])[C:8](\[C:9](=[C:10]1/[c:11]3[c:12]([c:13]([c:14]([c:15]([c:16]3[N:17]([C:18]1=[O:19])[H:20])[H:21])[H:22])[S:23](=[O:24])(=[O:25])[O-:26])[H:27])[N:28]2[H:29])=[O:30])[H:31])[H:32])[H:33] ZINC03814440|cdk2_ligands|dock2
[c:1]1([c:2]([c:3]([c:4]2[c:5]([c:6]1[H:7])[C:8](\[C:9](=[C:10]1/[c:11]3[c:12]([c:13]([c:14]([c:15]([c:16]3[N:17]([C:18]1=[O:19])[H:20])[H:21])[H:22])[S:23](=[O:24])(=[O:25])[O-:26])[H:27])[N:28]2[H:29])=[O:30])[H:31])[H:32])[H:33] ZINC03814440|cdk2_ligands|dock2_F#1
[c:1]1([c:2]([c:3]([c:4]([c:5]2[c:6]1[N:7]([C:8](\[C:9]2=[N:10]/[N:11]([c:12]2[c:13]([c:14]([c:15]([c:16]([c:17]2[H:18])[H:19])[S:20]([N:21]([H:22])[H:23])(=[O:24])=[O:25])[H:26])[H:27])[H:28])=[O:29])[H:30])[H:31])[Br:32])[H:33])[H:34] ZINC03814441|cdk2_ligands|dock5
[c:1]1([c:2]([c:3]([c:4]([c:5]2[c:6]1[N:7]([C:8](\[C:9]2=[N:10]/[N:11]([cH3:12])[H:28])=[O:29])[H:30])[H:31])[Br:32])[H:33])[H:34] ZINC03814441|cdk2_ligands|dock5_F#1
[c:12]1([NH2:11])[c:13]([c:14]([c:15]([c:16]([c:17]1[H:18])[H:19])[S:20]([N:21]([H:22])[H:23])(=[O:24])=[O:25])[H:26])[H:27] ZINC03814441|cdk2_ligands|dock5_F#2
[c:1]1([c:2]([c:3]([c:4]([c:5]2[c:6]1[N:7]([C:8](\[C:9]2=[N:10]/[N:11]([c:12]2[c:13]([c:14]([c:15]([c:16]([c:17]2[H:18])[H:19])[S:20]([N:21]([H:22])[H:23])(=[O:24])=[O:25])[H:26])[H:27])[H:28])=[O:29])[H:30])[H:31])[Br:32])[H:33])[H:34] ZINC03814441|cdk2_ligands|dock2
[c:1]1([c:2]([c:3]([c:4]([c:5]2[c:6]1[N:7]([C:8](\[C:9]2=[N:10]/[N:11]([cH3:12])[H:28])=[O:29])[H:30])[H:31])[Br:32])[H:33])[H:34] ZINC03814441|cdk2_ligands|dock2_F#1
[c:12]1([NH2:11])[c:13]([c:14]([c:15]([c:16]([c:17]1[H:18])[H:19])[S:20]([N:21]([H:22])[H:23])(=[O:24])=[O:25])[H:26])[H:27] ZINC03814441|cdk2_ligands|dock2_F#2
[c:1]1([c:2]([c:3]([c:4]([c:5]2[c:6]1[N:7]([C:8](\[C:9]2=[N:10]/[N:11]([c:12]2[c:13]([c:14]([c:15]([c:16]([c:17]2[H:18])[H:19])[S:20]([N:21]([H:22])[H:23])(=[O:24])=[O:25])[H:26])[H:27])[H:28])=[O:29])[H:30])[H:31])[Br:32])[H:33])[H:34] ZINC03814441|cdk2_ligands|dock4
[c:1]1([c:2]([c:3]([c:4]([c:5]2[c:6]1[N:7]([C:8](\[C:9]2=[N:10]/[N:11]([cH3:12])[H:28])=[O:29])[H:30])[H:31])[Br:32])[H:33])[H:34] ZINC03814441|cdk2_ligands|dock4_F#1
[c:12]1([NH2:11])[c:13]([c:14]([c:15]([c:16]([c:17]1[H:18])[H:19])[S:20]([N:21]([H:22])[H:23])(=[O:24])=[O:25])[H:26])[H:27] ZINC03814441|cdk2_ligands|dock4_F#2
[c:1]1([c:2]([c:3]([c:4]([c:5]([c:6]1[C:7]([S@@:8]([N:9]([C:10]([H:11])([H:12])[H:13])[H:14])(=[O:15])=[O:16])([H:17])[H:18])[H:19])[H:20])[N:21](\[N:22]=[C:23]1\[c:24]2[c:25]([c:26]([c:27]([c:28]3[c:29]2[s:30][c:31]([n:32]3)[H:33])[H:34])[H:35])[N:36]([C:37]1=[O:38])[H:39])[H:40])[H:41])[H:42] ZINC03814444|cdk2_ligands|dock6
[c:1]1([c:2]([c:3]([c:4]([c:5]([c:6]1[CH3:7])[H:19])[H:20])[NH2:21])[H:41])[H:42] ZINC03814444|cdk2_ligands|dock6_F#1
[C:7]([cH3:6])([S@@:8]([NH2:9])(=[O:15])=[O:16])([H:17])[H:18] ZINC03814444|cdk2_ligands|dock6_F#2
[N:9]([SH:8])([C:10]([H:11])([H:12])[H:13])[H:14] ZINC03814444|cdk2_ligands|dock6_F#3
[N:21]([cH3:3])(\[N:22]=[C:23]1\[c:24]2[c:25]([c:26]([c:27]([c:28]3[c:29]2[s:30][c:31]([n:32]3)[H:33])[H:34])[H:35])[N:36]([C:37]1=[O:38])[H:39])[H:40] ZINC03814444|cdk2_ligands|dock6_F#4
[c:1]1([c:2]([c:3]([c:4]([c:5]([c:6]1[C:7]([S@@:8]([N:9]([C:10]([H:11])([H:12])[H:13])[H:14])(=[O:15])=[O:16])([H:17])[H:18])[H:19])[H:20])[N:21](\[N:22]=[C:23]1\[c:24]2[c:25]([c:26]([c:27]([c:28]3[c:29]2[s:30][c:31]([n:32]3)[H:33])[H:34])[H:35])[N:36]([C:37]1=[O:38])[H:39])[H:40])[H:41])[H:42] ZINC03814444|cdk2_ligands|dock1
[c:1]1([c:2]([c:3]([c:4]([c:5]([c:6]1[CH3:7])[H:19])[H:20])[NH2:21])[H:41])[H:42] ZINC03814444|cdk2_ligands|dock1_F#1
[C:7]([cH3:6])([S@@:8]([NH2:9])(=[O:15])=[O:16])([H:17])[H:18] ZINC03814444|cdk2_ligands|dock1_F#2
[N:9]([SH:8])([C:10]([H:11])([H:12])[H:13])[H:14] ZINC03814444|cdk2_ligands|dock1_F#3
[N:21]([cH3:3])(\[N:22]=[C:23]1\[c:24]2[c:25]([c:26]([c:27]([c:28]3[c:29]2[s:30][c:31]([n:32]3)[H:33])[H:34])[H:35])[N:36]([C:37]1=[O:38])[H:39])[H:40] ZINC03814444|cdk2_ligands|dock1_F#4
[c:1]1([c:2]([c:3]([c:4]([c:5]([c:6]1[C:7]([S@@:8]([N:9]([C:10]([H:11])([H:12])[H:13])[H:14])(=[O:15])=[O:16])([H:17])[H:18])[H:19])[H:20])[N:21](\[N:22]=[C:23]1\[c:24]2[c:25]([c:26]([c:27]([c:28]3[c:29]2[s:30][c:31]([n:32]3)[H:33])[H:34])[H:35])[N:36]([C:37]1=[O:38])[H:39])[H:40])[H:41])[H:42] ZINC03814444|cdk2_ligands|dock10
[c:1]1([c:2]([c:3]([c:4]([c:5]([c:6]1[CH3:7])[H:19])[H:20])[NH2:21])[H:41])[H:42] ZINC03814444|cdk2_ligands|dock10_F#1
[C:7]([cH3:6])([S@@:8]([NH2:9])(=[O:15])=[O:16])([H:17])[H:18] ZINC03814444|cdk2_ligands|dock10_F#2
[N:9]([SH:8])([C:10]([H:11])([H:12])[H:13])[H:14] ZINC03814444|cdk2_ligands|dock10_F#3
[N:21]([cH3:3])(\[N:22]=[C:23]1\[c:24]2[c:25]([c:26]([c:27]([c:28]3[c:29]2[s:30][c:31]([n:32]3)[H:33])[H:34])[H:35])[N:36]([C:37]1=[O:38])[H:39])[H:40] ZINC03814444|cdk2_ligands|dock10_F#4
[c:1]1([c:2]([c:3]([n:4][c:5]([c:6]1[H:7])[N:8]([C:9]([N:10]([c:11]1[c:12]([c:13]([c:14]([c:15]2[c:16]1[C@@:17]1([N:18]([C:19]([C:20]([C:21]1([H:22])[H:23])([H:24])[H:25])([H:26])[H:27])[C:28]2=[O:29])[H:30])[H:31])[H:32])[H:33])[H:34])=[O:35])[H:36])[H:37])[H:38])[H:39] ZINC03814449|cdk2_ligands|dock2
[c:1]1([c:2]([c:3]([n:4][c:5]([c:6]1[H:7])[NH2:8])[H:37])[H:38])[H:39] ZINC03814449|cdk2_ligands|dock2_F#1
[N:8]([cH3:5])([CH3:9])[H:36] ZINC03814449|cdk2_ligands|dock2_F#2
[C:9]([NH2:8])([NH2:10])=[O:35] ZINC03814449|cdk2_ligands|dock2_F#3
[N:10]([CH3:9])([cH3:11])[H:34] ZINC03814449|cdk2_ligands|dock2_F#4
[c:11]1([NH2:10])[c:12]([c:13]([c:14]([c:15]2[c:16]1[C@@:17]1([N:18]([C:19]([C:20]([C:21]1([H:22])[H:23])([H:24])[H:25])([H:26])[H:27])[C:28]2=[O:29])[H:30])[H:31])[H:32])[H:33] ZINC03814449|cdk2_ligands|dock2_F#5
[c:1]1([c:2]([c:3]([n:4][c:5]([c:6]1[H:7])[N:8]([C:9]([N:10]([c:11]1[c:12]([c:13]([c:14]([c:15]2[c:16]1[C@@:17]1([N:18]([C:19]([C:20]([C:21]1([H:22])[H:23])([H:24])[H:25])([H:26])[H:27])[C:28]2=[O:29])[H:30])[H:31])[H:32])[H:33])[H:34])=[O:35])[H:36])[H:37])[H:38])[H:39] ZINC03814449|cdk2_ligands|dock5
[c:1]1([c:2]([c:3]([n:4][c:5]([c:6]1[H:7])[NH2:8])[H:37])[H:38])[H:39] ZINC03814449|cdk2_ligands|dock5_F#1
[N:8]([cH3:5])([CH3:9])[H:36] ZINC03814449|cdk2_ligands|dock5_F#2
[C:9]([NH2:8])([NH2:10])=[O:35] ZINC03814449|cdk2_ligands|dock5_F#3
[N:10]([CH3:9])([cH3:11])[H:34] ZINC03814449|cdk2_ligands|dock5_F#4
[c:11]1([NH2:10])[c:12]([c:13]([c:14]([c:15]2[c:16]1[C@@:17]1([N:18]([C:19]([C:20]([C:21]1([H:22])[H:23])([H:24])[H:25])([H:26])[H:27])[C:28]2=[O:29])[H:30])[H:31])[H:32])[H:33] ZINC03814449|cdk2_ligands|dock5_F#5
[c:1]1([c:2]([c:3]([n:4][c:5]([c:6]1[H:7])[N:8]([C:9]([N:10]([c:11]1[c:12]([c:13]([c:14]([c:15]2[c:16]1[C@@:17]1([N:18]([C:19]([C:20]([C:21]1([H:22])[H:23])([H:24])[H:25])([H:26])[H:27])[C:28]2=[O:29])[H:30])[H:31])[H:32])[H:33])[H:34])=[O:35])[H:36])[H:37])[H:38])[H:39] ZINC03814449|cdk2_ligands|dock7
[c:1]1([c:2]([c:3]([n:4][c:5]([c:6]1[H:7])[NH2:8])[H:37])[H:38])[H:39] ZINC03814449|cdk2_ligands|dock7_F#1
[N:8]([cH3:5])([CH3:9])[H:36] ZINC03814449|cdk2_ligands|dock7_F#2
[C:9]([NH2:8])([NH2:10])=[O:35] ZINC03814449|cdk2_ligands|dock7_F#3
[N:10]([CH3:9])([cH3:11])[H:34] ZINC03814449|cdk2_ligands|dock7_F#4
[c:11]1([NH2:10])[c:12]([c:13]([c:14]([c:15]2[c:16]1[C@@:17]1([N:18]([C:19]([C:20]([C:21]1([H:22])[H:23])([H:24])[H:25])([H:26])[H:27])[C:28]2=[O:29])[H:30])[H:31])[H:32])[H:33] ZINC03814449|cdk2_ligands|dock7_F#5